components/scream/cime_config/buildnml

```diff
@@ -176,304 +278,106 @@ def evaluate_selectors(element, case, ez_selectors):
-    ...       <baz hgrid="no_match" nlev="128">2</baz>
-    ...       <bat>1</bat>
-    ...       <bat nlev="128" hgrid="no_match">2</bat>
-    ...       <sel_regex_no_match>1</sel_regex_no_match>
-    ...       <sel_regex_no_match nomatch=".*">2</sel_regex_no_match>
-    ...     </group2>
-    ...   </file>
-    ... </namelist_defaults>
-    ... '''
-    >>> print(_create_raw_xml_file_impl(case, full_test)[0])
-    <namelist_defaults>
-      <file name="test.yaml" format="yaml">
-        <group1>
-          <testval>42</testval>
-          <testval2>1234</testval2>
-          <interleave>hi</interleave>
-          </group1>
-        <group2>
-          <testval>42</testval>
-          <testval2>1234</testval2>
-          <interleave>hi</interleave>
-          <prec_tester>3</prec_tester>
-          <foo>2</foo>
-          <bar>1</bar>
-          <baz>1</baz>
-          <bat>1</bat>
-          <sel_regex_no_match>1</sel_regex_no_match>
-          </group2>
-      </file>
-    </namelist_defaults>
-    >>> import pprint
-    >>> pp = pprint.PrettyPrinter(indent=4)
-    >>> pp.pprint(_create_raw_xml_file_impl(case, full_test)[1])
-    {   'test.yaml': {   'group1': {   'interleave': ['string', None],
-                                       'testval': ['integer', None],
-                                       'testval2': ['integer', None]},
-                         'group2': {   'bar': ['integer', None],
-                                       'bat': ['integer', None],
-                                       'baz': ['integer', None],
-                                       'foo': ['integer', None],
-                                       'interleave': ['string', None],
-                                       'prec_tester': ['integer', None],
-                                       'sel_regex_no_match': ['integer', None],
-                                       'testval': ['integer', None],
-                                       'testval2': ['integer', None]}}}
-    >>> ######################################################################
-    >>> test_metadata_override = '''
+    On input, xml contains the parsed content of namelist_defaults_scream.xml.
+    On output, it contains the input parameters for this case.
+
+    >>> from eamxx_buildnml_impl import MockCase
+    >>> case = MockCase({'foo':1, 'bar':2, 'baz_bar':'blah', 'baz2_bar2':4})
+    >>> xml = '''
     ... <namelist_defaults>
-    ...   <selectors>
-    ...     <selector name="hgrid" case_env="ATM_GRID"/>
-    ...   </selectors>
-    ...   <file name="test.yaml" format="yaml">
-    ...     <testval valid_values="1,2,3">1</testval>
-    ...     <testval hgrid="ne4ne4" valid_values="1,2,3,4">2</testval>
-    ...     <testval hgrid="no_match" valid_values="1,2,3,4,5">3</testval>
-    ...     <testval2 valid_values="1,2,3">1</testval2>
-    ...     <testval2 hgrid="ne4ne4">2</testval2>
-    ...     <testval2 hgrid="no_match">3</testval2>
-    ...   </file>
+    ...     <selectors/>
```

Comment:
This is a good new test, but I still don't understand why some of the tests above were deleted instead of adapted to work with this new iteration? I can see some of the selector tests were moved but it looks like some were just lost.